resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/NullFileSystem.java

```diff
@@ -27,34 +27,29 @@
 public class NullFileSystem extends FileSystem {
 
 	/**
-	 * The singleton instance of this file system.
+	 * Creates the null file system.
 	 */
-	private static IFileSystem instance;
+	public NullFileSystem() {
+		super();
+	}
 
 	/**
 	 * Returns the instance of this file system
 	 *
 	 * @return The instance of this file system.
 	 */
+	@Deprecated(forRemoval = true)
```

Comment:
When you deprecate something with forRemova=true please add since with proper value e.g. "2024-03" so it's known when it can be removed.

Reply:
@akurtakov it is internal api so it can be removed anytime....

Reply:
I missed that. I'm stepping back here :)

Reply:
It was only @iloveeclipse was concerned that someone somewhere is using our internal API even though as it obviously never has worked reliable...